clang/test/CodeGen/arithmetic-fence-builtin.c

```diff
@@ -1,20 +1,30 @@
 // Test with fast math
 // RUN: %clang_cc1 -triple i386-pc-linux-gnu -emit-llvm -DFAST \
 // RUN: -mreassociate \
-// RUN: -o - %s | FileCheck --check-prefixes CHECK,CHECKFAST,CHECKNP %s
+// RUN: -o - %s | FileCheck --check-prefixes CHECK,CHECKFAST,CHECKPREC,CHECKNP %s
 //
 // Test with fast math and fprotect-parens
 // RUN: %clang_cc1 -triple i386-pc-linux-gnu -emit-llvm -DFAST \
 // RUN: -mreassociate -fprotect-parens -ffp-contract=on\
-// RUN: -o - %s | FileCheck --check-prefixes CHECK,CHECKFAST,CHECKPP %s
+// RUN: -o - %s | FileCheck --check-prefixes CHECK,CHECKFAST,CHECKPREC,CHECKPP %s
 //
 // Test without fast math: llvm intrinsic not created
 // RUN: %clang_cc1 -triple i386-pc-linux-gnu -emit-llvm -fprotect-parens\
 // RUN: -o - %s | FileCheck --implicit-check-not="llvm.arithmetic.fence" %s
 //
-int v;
-int addit(float a, float b) {
-  // CHECK: define {{.*}}@addit(float noundef %a, float noundef %b) #0 {
+// RUN: %clang_cc1 -triple spir64-unknown-unknown  -emit-llvm -fsycl-is-device \
+// RUN: -disable-llvm-passes -mreassociate -opaque-pointers -DSYCLD \
```

Comment:
Is `disable-llvm-passes` flag still necessary? I would think after we switched to -O0 by default for SYCL it is not.